src/hierarchies/updaters/conjugate_updater.h

```diff
@@ -0,0 +1,57 @@
+#ifndef BAYESMIX_HIERARCHIES_UPDATERS_CONJUGATE_UPDATER_H_
+#define BAYESMIX_HIERARCHIES_UPDATERS_CONJUGATE_UPDATER_H_
+
+#include "abstract_updater.h"
+#include "src/hierarchies/likelihoods/abstract_likelihood.h"
+#include "src/hierarchies/priors/abstract_prior_model.h"
+
+template <class Likelihood, class PriorModel>
+class ConjugateUpdater : public AbstractUpdater {
```

Comment:
I would rename this `SemiConjugateUpdater`, since it works also for semi-conjugate hierarchies